modules/core/keeper/keeper.go

```diff
@@ -46,6 +49,15 @@ func NewKeeper(
 		paramSpace = paramSpace.WithKeyTable(keyTable)
 	}
 
+	// panic if any of the keepers passed in is empty
+	if reflect.ValueOf(stakingKeeper).IsZero() {
+		panic(fmt.Errorf("cannot initialize ibc keeper: empty staking keeper"))
+	} else if reflect.ValueOf(upgradeKeeper).IsZero() {
+		panic(fmt.Errorf("cannot initialize ibc keeper: empty upgrade keeper"))
+	} else if reflect.DeepEqual(capabilitykeeper.ScopedKeeper{}, scopedKeeper) {
+		panic(fmt.Errorf("cannot initialize ibc keeper: empty scoped keeper"))
+	}
```

Comment:
super nit (you don't need to apply it)
```suggestion
	if reflect.ValueOf(stakingKeeper).IsZero() {
		panic(fmt.Errorf("cannot initialize IBC keeper: empty staking keeper"))
	} 
	
	if reflect.ValueOf(upgradeKeeper).IsZero() {
		panic(fmt.Errorf("cannot initialize IBC keeper: empty upgrade keeper"))
	} 
	
	if reflect.DeepEqual(capabilitykeeper.ScopedKeeper{}, scopedKeeper) {
		panic(fmt.Errorf("cannot initialize IBC keeper: empty scoped keeper"))
	}
```

I find this easier to read

Reply:
Hey @catShaark, I also find this much easier to read and is probably considered more idiomatic Go, since the flow of execution will terminate with `panic()` I think a single conditional per keeper type is cleaner!
Personally I'm in favour of applying the suggested code snippet! :)

Reply:
yeah, it does look cleaner and I just applied the suggested change